df_py/volume/reward_calculator.py

```diff
@@ -0,0 +1,479 @@
+          df_week -- int DF_week
+          OCEAN_avail -- amount of rewards avail, in units of OCEAN
+          do_pubrewards -- 2x effective stake to publishers?
+          do_rank -- allocate OCEAN to assets by DCV rank, vs pro-rata
+        """
+        self._freeze_attributes = False
+
+        self.stakes = cc.mod_stakes(stakes)
+        self.nftvols = cc.mod_nft_vols(nftvols)
+        self.owners = cc.mod_owners(owners)
+        self.symbols = cc.mod_symbols(symbols)
+        self.rates = cc.mod_rates(rates)
+
+        self.nftvols_USD = to_usd.nft_vols_to_usd(
+            self.nftvols, self.symbols, self.rates
+        )
+
+        self.chain_nft_tups = self._get_chain_nft_tups()
+        self.LP_addrs = self._get_lp_addrs()
+
+        self.df_week = df_week
+        self.OCEAN_avail = OCEAN_avail
+        self.do_pubrewards = do_pubrewards
+        self.do_rank = do_rank
+
+        self.predictoors = self._get_predictoors()
+
+        self._freeze_attributes = True
+
+    @enforce_types
+    def calculate(self):
+        """
+        @notes
+          In the return dicts, chainID is the chain of the nft, not the
+          chain where rewards go.
+        """
+        self._freeze_attributes = False
+
+        self.S, self.V_USD, self.M, self.C = self._stake_vol_owner_dicts_to_arrays()
+        self.R = self._calc_usd()
+
+        self._freeze_attributes = True
+
+        (rewardsperlp, rewardsinfo) = self._reward_array_to_dicts()
+
+        return rewardsperlp, rewardsinfo
+
+    @freeze_attributes
+    @enforce_types
+    def _stake_vol_owner_dicts_to_arrays(
+        self,
+    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+        """
+        @return
+          S -- 2d array of [LP i, chain_nft j] -- stake for each {i,j}, in veOCEAN
+          V_USD -- 1d array of [chain_nft j] -- nftvol for each {j}, in USD
+        """
+        N_j = len(self.chain_nft_tups)
+        N_i = len(self.LP_addrs)
+
+        M = np.zeros(N_j, dtype=float)
+        S = np.zeros((N_i, N_j), dtype=float)
+        V_USD = np.zeros(N_j, dtype=float)
+        C = np.zeros(N_j, dtype=int)
+
+        for j, (chainID, nft_addr) in enumerate(self.chain_nft_tups):
+            for i, LP_addr in enumerate(self.LP_addrs):
+                assert nft_addr in self.stakes[chainID], "each tup should be in stakes"
+                S[i, j] = self.stakes[chainID][nft_addr].get(LP_addr, 0.0)
+            V_USD[j] += self.nftvols_USD[chainID].get(nft_addr, 0.0)
+
+            M[j] = calc_dcv_multiplier(
+                self.df_week, nft_addr in self.predictoors[chainID]
+            )
+
+            owner_addr = self.owners[chainID][nft_addr]
+            C[j] = (
+                -1
+                if owner_addr not in self.LP_addrs
+                else self.LP_addrs.index(owner_addr)
+            )
+
+        return S, V_USD, M, C
+
+    @freeze_attributes
+    @enforce_types
+    def _calc_usd(self) -> np.ndarray:
```

Comment:
Rename to _calc_rewards_usd().

(It should have always been this, but now's a great time to do so.)

Reply:
it was like that... I renamed it because I needed some clarity, but I renamed it back :)